manager/job/sync_peers.go

```diff
@@ -55,24 +57,92 @@ type syncPeers struct {
 	job    *internaljob.Job
 	db     *gorm.DB
 	done   chan struct{}
+
+	syncLocker sync.Mutex
+	workChan   chan SyncPeersArgs
+}
+
+type SyncPeersArgs struct {
+	CandidateSchedulerClusters []models.SchedulerCluster
+	TaskID                     string
 }
 
 // newSyncPeers returns a new SyncPeers.
 func newSyncPeers(cfg *config.Config, job *internaljob.Job, gdb *gorm.DB) (SyncPeers, error) {
 	return &syncPeers{
-		config: cfg,
-		db:     gdb,
-		job:    job,
-		done:   make(chan struct{}),
+		config:     cfg,
+		db:         gdb,
+		job:        job,
+		done:       make(chan struct{}),
+		workChan:   make(chan SyncPeersArgs),
+		syncLocker: sync.Mutex{},
 	}, nil
 }
 
-// Run sync peers.
-func (s *syncPeers) Run(ctx context.Context) error {
-	// Find all of the scheduler clusters that has active schedulers.
-	var candidateSchedulerClusters []models.SchedulerCluster
-	if err := s.db.WithContext(ctx).Find(&candidateSchedulerClusters).Error; err != nil {
-		return err
+// Run start to sync peers.
+func (s *syncPeers) Run(ctx context.Context, args SyncPeersArgs) error {
+	if len(args.CandidateSchedulerClusters) == 0 {
+		if err := s.db.WithContext(ctx).Find(&args.CandidateSchedulerClusters).Error; err != nil {
+			return fmt.Errorf("failed to get candidate scheduler clusters: %v", err)
+		}
+	}
+
+	s.workChan <- args
```

Comment:
As the comment described in the Run method, it should be async function, but it will hang out here as it is a no buffered channel.

Reply:
Although the subsequent logic in the code quickly processes this channel, I do believe that adding a buffer would be better. Set the size as 10.